llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -364,14 +364,15 @@ unsigned GCNTTIImpl::getStoreVectorFactor(unsigned VF, unsigned StoreSize,
 }
 
 unsigned GCNTTIImpl::getLoadStoreVecRegBitWidth(unsigned AddrSpace) const {
-  if (AddrSpace == AMDGPUAS::GLOBAL_ADDRESS ||
-      AddrSpace == AMDGPUAS::CONSTANT_ADDRESS ||
+  if (AddrSpace == AMDGPUAS::GLOBAL_ADDRESS)
+    return 512;
```

Comment:
128 would be more correct on average for global than constant

Reply:
This patch does not have any effect for GLOBAL_ADDRESS. Is the suggestion to also modify the value returned for this case to 128? (I haven't investigated that).

Reply:
I also note the values returned for R600 (R600TTIImpl::getLoadStoreVecRegBitWidth) are all not larger than 128.

Reply:
R600 didn't have the really wide scalar loads